### PR TITLE
Update addons yaml files for converting tolerations to api fields.

### PR DESCRIPTION
**Pull request by aveshagarwal**

### CODE REVIEW

**Comment by aveshagarwal** [comment]
It seems that https://github.com/kubernetes/kops/ needs to be updated as it is using scheduler.alpha.kubernetes.io/taints: '[{"key":"dedicated","value":"master","effect":"NoSchedule"}]' for tainting nodes during tests.

**Comment by piosz** [comment]
LGTM, thought this should be done together with bumping Rescheduler version. I'll ping you once the rescheduler change is ready.

**Comment by aveshagarwal** [comment]
@piosz thanks. I think this PR can not be merged untill addon manager is fixed, otherwise several e2e tests will fail. @MrHohn

**Comment by aveshagarwal** [comment]
@piosz @MrHohn since the pr (https://github.com/kubernetes/kubernetes/pull/38957) that changed taints/tolerations to api field is already merged now, I have rebased this PR too.

**Comment by MrHohn** [comment]
@aveshagarwal Good to know that, I'm pushing new addon-manager image to upgrade its kubectl binary.

**Comment by davidopp** [comment]
@aveshagarwal  I thought you updated the YAML files in your main PR (the one that already merged)? What is this PR doing?

**Comment by aveshagarwal** [comment]
@davidopp These addon yaml files can only be updated if addon manager has been updated. There is PR to update addon manager here: https://github.com/kubernetes/kubernetes/issues/42017. Once that PR merges, then I am assuming all tests in this PR should pass. and then this PR can be merged.

**Comment by davidopp** [comment]
Ah OK, thanks

**Comment by MrHohn** [comment]
Sorry for blocking this PR. The addon-manager is now updating by #40108 instead and I will close #42017.

**Comment by timothysc** [comment]
Someone who owns the kube-up scriptery should approve. 

/cc @kubernetes/sig-scheduling-misc

**Comment by davidopp** [comment]
@k8s-bot test this

File: cluster/addons/dns/kubedns-controller.yaml.in [hunk 1]
```diff
@@ -43,8 +43,10 @@ spec:
         k8s-app: kube-dns
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ''
-        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}]'
     spec:
+      toleration:
```
Comment (MrHohn) [inline cluster/addons/dns/kubedns-controller.yaml.in]: s/toleration/tolerations

This is causing addon-manager failures:
```
error: error validating "/etc/kubernetes/addons/dns/kubedns-controller.yaml": error validating data: found invalid field toleration for v1.PodSpec
```

Comment (aveshagarwal) [inline cluster/addons/dns/kubedns-controller.yaml.in]: ok looking into it.

**Comment by aveshagarwal** [comment]
@davidopp @MrHohn fixed it. I dont have rights to set flags again, could one of you do it after tests pass?

**Comment by aveshagarwal** [comment]
@piosz @MrHohn @davidopp pinging again as all tests passed so ready for lgtm/approve I think.

**Pull request merged**